scripts/pylib/twister/twisterlib/harness.py

```diff
@@ -375,43 +375,43 @@ def _update_test_status(self):
                 self._parse_report_file(self.report_file)
             except Exception as e:
                 logger.error(f'Error when parsing file {self.report_file}: {e}')
-                self.state = 'failed'
+                self.state = Status.FAIL
             finally:
                 if not self.instance.testcases:
                     self.instance.init_cases()
 
-        self.instance.status = self.state or 'failed'
-        if self.instance.status in ['error', 'failed']:
+        self.instance.status = self.state or Status.FAIL
+        if self.instance.status in [Status.ERROR, Status.FAIL]:
             self.instance.reason = self.instance.reason or 'Pytest failed'
-            self.instance.add_missing_case_status('blocked', self.instance.reason)
+            self.instance.add_missing_case_status(Status.BLOCK, self.instance.reason)
 
     def _parse_report_file(self, report):
         tree = ET.parse(report)
         root = tree.getroot()
         if elem_ts := root.find('testsuite'):
             if elem_ts.get('failures') != '0':
-                self.state = 'failed'
+                self.state = Status.FAIL
             elif elem_ts.get('errors') != '0':
-                self.state = 'error'
-            elif elem_ts.get('skipped') == elem_ts.get('tests'):
-                self.state = 'skipped'
+                self.state = Status.ERROR
+            elif elem_ts.get(Status.SKIP) == elem_ts.get('tests'):
+                self.state = Status.SKIP
             else:
-                self.state = 'passed'
+                self.state = Status.PASS
             self.instance.execution_time = float(elem_ts.get('time'))
 
             for elem_tc in elem_ts.findall('testcase'):
                 tc = self.instance.get_case_or_create(f"{self.id}.{elem_tc.get('name')}")
                 tc.duration = float(elem_tc.get('time'))
                 elem = elem_tc.find('*')
                 if elem is None:
-                    tc.status = 'passed'
+                    tc.status = Status.PASS
                 else:
-                    if elem.tag == 'skipped':
-                        tc.status = 'skipped'
+                    if elem.tag == Status.SKIP:
```

Comment:
Better not to compare two different objects. `elem.tag` is a literal tag name from parsing junit xml results from pytest. For now, there is the same string used, but it could be changed.